src/testCommon/java/uk/gov/hmcts/darts/test/common/data/HearingTestData.java

```diff
@@ -50,7 +50,10 @@ public static HearingEntity someMinimalHearing() {
     }
 
     public static HearingEntity createHearingWith(CourtCaseEntity courtCase, CourtroomEntity courtroom, LocalDate hearingDate) {
-        return createHearingWithDefaults(courtCase, courtroom, hearingDate, null);
+        var hearing = createHearingFor(courtCase);
```

Comment:
`createHearingFor` calls `someMinimalHearing()` which in turn creates its own case, courtroom etc, yet we are then instantly overriding them. Could we do something similar like what i mentioned about Prosecutor?
I notice you removed the call to `createHearingWithDefaults`, curious as to why it caused problems.

Reply:
same a above